app/nagbot.py

```diff
@@ -116,11 +116,15 @@ def execute_internal(channel, dryrun):
             if len(resources_to_terminate) > 0:
                 message = f'I terminated the following {ec2_type}s: '
                 for r in resources_to_terminate:
-                    contact = sqslack.lookup_user_by_email(r.contact)
-                    message = message + r.make_resource_summary() + \
-                        f', "Terminate after"={r.terminate_after}, "Monthly Price"=' \
-                        f'{util.money_to_string(r.monthly_price)}, Contact={contact}\n'
-                    r.terminate_resource(dryrun=dryrun)
+                    response = r.terminate_resource(dryrun=dryrun)
+                    if response is not True:
```

Comment:
`response` can be an `Exception` type and IMHO it is not ideal to compare an Exception to a bool

Reply:
The `terminate_resource()` method has to return a string containing the error message if there is an error so that message can be included in the slack output, but I suppose it doesn't necessarily have to return a boolean `True` if the termination is successful. I we wish to keep the return types of the terminate method the same regardless of success or failure, we could return the string 'true' if the termination is successful rather than a Boolean, so that the return value is always a string. If success and failure both resulted in a string, the logic could be changed to `if response == "true"`; however, I prefer to leave it as is and return a boolean `True' upon success, and a string with the error message upon failure - I don't anticipate any issues that could arise from the comparison in this case, but if there could be negative implications from the comparison i'm overlooking, I can make the change.

Reply:
I agree that Exception and bool are not a great combo to return, i'd suggest using `None/String` or a list of errors. 

You can also effectively return two values in python

```
def terminate():
  return True, error_str

success, errors = terminate()
```

Reply:
I like the idea of using a `None` / `String` combo as return values.

A list of errors would make sense for the AMI `terminate()` method, while a None / String combo would make more sense for the other resources since there would only be one error in those cases. 

In order to keep the return value of the `terminate()` method of each resource the same, i'll use the None / String method for all of them.